packages/flutter/lib/src/rendering/view.dart

```diff
@@ -184,6 +186,14 @@ class RenderView extends RenderObject with RenderObjectWithChildMixin<RenderBox>
     }
   }
 
+  void _updateSystemChrome() {
+    final SystemUiOverlayStyle overlayStyle = layer.findRegion(
+      Offset.zero,
```

Comment:
The suggestion I had was for when we just had one colour for everything. Now that we have two colours (at least), we probably want to independently get the details for the top and the bottom.

For the top, I'd probably aim for a point half way across the width, and half of the way down the top padding.
Similarly for the other sources.